core/src/com/fourinachamber/fortyfive/game/controller/NewGameController.kt

```diff
@@ -0,0 +1,368 @@
+    override val screen: OnjScreen,
+    val gameEvents: EventPipeline
+) : ScreenController(), GameController {
+
+    private val gameDirector = GameDirector(this)
+
+    override val gameRenderPipeline: GameRenderPipeline
+        get() = TODO("Not yet implemented")
+
+    override val playerLost: Boolean
+        get() = TODO("Not yet implemented")
+
+    override var curReserves: Int = Config.baseReserves
+
+    override val isUIFrozen: Boolean
+        get() = TODO("Not yet implemented")
+
+    override val revolverRotationCounter: Int
+        get() = TODO("Not yet implemented")
+
+    override val turnCounter: Int
+        get() = TODO("Not yet implemented")
+
+    override val playerStatusEffects: List<StatusEffect>
+        get() = TODO("Not yet implemented")
+
+    override val isEverlastingDisabled: Boolean
+        get() = TODO("Not yet implemented")
+
+    override val cardsInHand: List<Card>
+        get() = TODO("Not yet implemented")
+
+    private val _encounterModifiers: MutableList<EncounterModifier> = mutableListOf()
+    override val encounterModifiers: List<EncounterModifier>
+        get() = _encounterModifiers
+
+    override val curPlayerLives: Int
+        get() = TODO("Not yet implemented")
+
+    override val activeEnemies: List<Enemy>
+        get() = TODO("Not yet implemented")
+
+    override val allEnemies: List<Enemy>
+        get() = TODO("Not yet implemented")
+
+    override val shootButton: Actor
+        get() = TODO("Not yet implemented")
+
+    @Inject override lateinit var revolver: Revolver
+
+    @Inject private lateinit var cardHand: NewCardHand
+
+    override lateinit var encounterContext: EncounterContext
+        private set
+
+    private var cardPrototypes: List<CardPrototype> = listOf()
+
+    private val _cardStack: MutableList<Card> = mutableListOf()
+    override val cardStack: List<Card>
+        get() = _cardStack
+
+    private val createdCards: MutableList<Card> = mutableListOf()
+
+    private lateinit var defaultBullet: CardPrototype
+
+    private val mainTimeline: Timeline = Timeline().also { it.startTimeline() }
+    private val animTimelines: MutableList<Timeline> = mutableListOf()
+
+    private val tutorialText: MutableList<GameDirector.GameTutorialTextPart> = mutableListOf()
+
+    override fun init(context: Any?) {
+        if (context !is EncounterContext) {
+            throw RuntimeException("GameScreen needs a context of type encounterMapEvent")
+        }
+        encounterContext = context
+        FortyFive.currentGame = this
+
+        gameDirector.init()
+
+        gameEvents.watchFor<NewCardHand.CardDraggedOntoSlotEvent> { loadBulletFromHandInRevolver(it.card, it.slot.num) }
+
+        initCards()
+        _cardStack.forEach { cardHand.addCard(it) }
+        updateReserves(Config.baseReserves)
+    }
+
+    private fun updateReserves(newReserves: Int, sourceActor: Actor? = null) {
+        val prevReserves = curReserves
+        curReserves = newReserves
```

Comment:
check if the reserves actually changed, and newReserves is not just the same as curReserves